src/test/java/uk/gov/hmcts/reform/userprofileapi/util/IdamStatusResolverTest.java

```diff
@@ -83,15 +94,90 @@ public IdamRolesInfo createIdamRoleInfo(Boolean active, Boolean pending) {
         assertFalse(constructor.isAccessible());
         constructor.setAccessible(true);
         constructor.newInstance((Object[]) null);
     }
+
+    @Test
+    public void test_resolveStatusAndReturnMessage_when_responseEntity_is_null() {
+        ResponseEntity<Object> responseEntity = null;
+        String errorMessage = resolveStatusAndReturnMessage(responseEntity);
+        assertThat(errorMessage).isEqualTo(resolveStatusAndReturnMessage(INTERNAL_SERVER_ERROR));
+    }
+
+    @Test
+    public void test_resolveStatusAndReturnMessage_when_responseEntity_body_is_null() {
+        ResponseEntity<Object> responseEntity = ResponseEntity.status(NOT_FOUND).build();
+        String errorMessage = resolveStatusAndReturnMessage(responseEntity);
+        assertThat(errorMessage).isEqualTo(resolveStatusAndReturnMessage(NOT_FOUND));
+    }
+
+    @Test
+    public void test_resolveStatusAndReturnMessage_when_responseEntity_body_has_not_instance_of_IdamErrorResponse() {
+        ResponseEntity<Object> responseEntity = ResponseEntity.status(CREATED).body(new ErrorResponse());
+        String errorMessage = resolveStatusAndReturnMessage(responseEntity);
+        assertThat(errorMessage).isEqualTo(resolveStatusAndReturnMessage(CREATED));
+    }
+
+    @Test
+    public void test_resolveStatusAndReturnMessage_when_responseEntity_body_has_IdamErrorResponse() {
+        IdamErrorResponse idamErrorResponse = new IdamErrorResponse();
+        idamErrorResponse.setStatus(400);
+        idamErrorResponse.setErrorMessage("some test error message");
+        ResponseEntity<Object> responseEntity = ResponseEntity.status(CREATED).body(idamErrorResponse);
+        String errorMessage = resolveStatusAndReturnMessage(responseEntity);
+        assertThat(errorMessage).isEqualTo("some test error message");
+    }
+
+    @Test
+    public void test_getErrorMessageFromSidamResponse_when_responseEntity_body_has_IdamErrorResponse_with_all_fields() {
+        List<String> errorMessages = new ArrayList<>();
+        errorMessages.add("errorMessage1");
+        errorMessages.add("errorMessage2");
+        IdamErrorResponse idamErrorResponse = new IdamErrorResponse();
+        idamErrorResponse.setStatus(400);
+        idamErrorResponse.setErrorMessage("some test error message");
+        idamErrorResponse.setErrorMessages(errorMessages);
+        String errorMessage = getErrorMessageFromSidamResponse(idamErrorResponse);
+        assertThat(errorMessage).isEqualTo("errorMessage1");
+    }
+
+    @Test
+    public void test_getErrorMessageFromSidamResponse_when_responseEntity_body_has_IdamErrorResponse_with_ErrorMessages_field() {
+        List<String> errorMessages = new ArrayList<>();
+        errorMessages.add("errorMessage1");
+        errorMessages.add("errorMessage2");
+        IdamErrorResponse idamErrorResponse = new IdamErrorResponse();
+        idamErrorResponse.setStatus(400);
+        idamErrorResponse.setErrorMessages(errorMessages);
```

Comment:
same code duplicated in the each test case ..could you please move to the common code to private method and reuse in the test methods
IdamErrorResponse idamErrorResponse = new IdamErrorResponse();
        idamErrorResponse.setStatus(400);
        idamErrorResponse.setErrorMessages(errorMessages);
        String errorMessage = getErrorMessageFromSidamResponse(idamErrorResponse);

Reply:
done